File: day-4/string_manipulation.py
class StringManipulation:

    def swapcase(self, s: str) -> str:
        if not s:
            return s
        else:
            return self.swapchar(s[0]) + self.swapcase(s[1:])

    def swapchar(self, s: str) -> str:
        if s in 'abcdefghijklmnopqrstuvwxyz':
            return s.upper()
        else:
            return s.lower()


if __name__ == '__main__':
    string_manipulation = StringManipulation()
    print(string_manipulation.swapcase('hEllO wOrLd'))
    assert 'HeLLo WoRlD' == string_manipulation.swapcase('hEllO wOrLd')

File: day-4/string_manipulation.clj
(defn toggle-case [s]
  (if (= (clojure.string/upper-case s) (str s))
    (clojure.string/lower-case s)
    (clojure.string/upper-case s)))

(defn swap-case [s]
  (if (not (empty? s))
    (str (toggle-case (first s)) (swap-case (rest s)))
    ""))

(println (swap-case "hEllO wOrLd"))
